// @tutors-sdk/tutors-lib/src/models/lo.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const path = __importStar(require("path"));
const futils_1 = require("../utils/futils");
const fs = __importStar(require("fs"));
const videoutils_1 = require("../utils/videoutils");
class LearningObject {
    constructor(parent) {
        this.hide = false;
        if (parent) {
            this.parent = parent;
        }
        this.lotype = 'lo';
    }
    reap(pattern) {
        this.folder = path.basename(process.cwd());
        this.parentFolder = futils_1.getParentFolder();
        this.img = futils_1.getImageFile(pattern);
        if (fs.existsSync('properties.yaml')) {
            this.properties = futils_1.readYaml('properties.yaml');
        }
        if (fs.existsSync(pattern + '.md')) {
            this.title = futils_1.getHeader(pattern + '.md');
            this.title = this.title + ' ';
            this.objectivesMd = futils_1.withoutHeader(pattern + '.md');
        }
        else {
            this.title = pattern;
        }
        this.videoids = videoutils_1.readVideoIds();
        this.videoid = this.videoids.videoid;
    }
}
exports.LearningObject = LearningObject;
