// @tutors-sdk/tutors-lib/src/models/topic.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const lo_1 = require("./lo");
const loutils_1 = require("../utils/loutils");
const fs = __importStar(require("fs"));
const sh = __importStar(require("shelljs"));
const futils_1 = require("../utils/futils");
class Topic extends lo_1.LearningObject {
    constructor(parent) {
        super(parent);
        this.los = [];
        this.allLos = [];
        this.los = loutils_1.reapLos({ parent: this });
        this.reap('topic');
        this.link = 'index.html';
        this.lotype = 'topic';
        this.setDefaultImage();
        this.los.forEach((lo) => this.allLos.push(lo));
        this.units = this.los.filter((lo) => lo.lotype == 'unit');
        this.panelVideos = this.los.filter((lo) => lo.lotype == 'panelvideo');
        this.panelTalks = this.los.filter((lo) => lo.lotype == 'paneltalk');
        this.standardLos = this.los.filter((lo) => lo.lotype !== 'unit' && lo.lotype !== 'panelvideo' && lo.lotype !== 'paneltalk');
    }
    setDefaultImage() {
        if (!this.img && this.los.length > 0) {
            const img = this.los[0].folder + '/' + this.los[0].img;
            if (fs.existsSync(img)) {
                this.img = img;
            }
        }
    }
    publish(path) {
        console.log('::', this.title);
        sh.cd(this.folder);
        const topicPath = path + '/' + this.folder;
        futils_1.copyFileToFolder(this.img, topicPath);
        loutils_1.publishLos(topicPath, this.los);
        sh.cd('..');
    }
}
exports.Topic = Topic;
class Unit extends Topic {
    constructor(parent) {
        super(parent);
        this.standardLos = [];
        this.lotype = 'unit';
        this.standardLos = this.los.filter((lo) => lo.lotype !== 'panelvideo');
    }
}
exports.Unit = Unit;
